File: backend/src/main/java/com/backend/backend/dao/entities/AudioCommentaire.java
package com.backend.backend.dao.entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.util.Date;

@Entity
@Table(name = "audio_commentaires")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AudioCommentaire {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String cheminFichier;

    private Integer dureeSecondes;

    @Temporal(TemporalType.TIMESTAMP)
    private Date dateEnregistrement;

    // Transcription via Whisper (null si pas encore transcrit)
    @Lob
    @Column(columnDefinition = "TEXT")
    private String transcription;

    @ManyToOne
    @JoinColumn(name = "fuite_id", nullable = false)
    private Fuite fuite;
}

File: frontend/lib/models/audio_commentaire.dart
class AudioCommentaire {
  final int id;
  final int fuiteId;
  final String cheminFichier;
  final int? dureeSecondes;
  final String? dateEnregistrement;
  final String? transcription;

  AudioCommentaire({
    required this.id,
    required this.fuiteId,
    required this.cheminFichier,
    this.dureeSecondes,
    this.dateEnregistrement,
    this.transcription,
  });

  factory AudioCommentaire.fromJson(Map<String, dynamic> json) {
    return AudioCommentaire(
      id: json['id'] as int,
      fuiteId: json['fuiteId'] as int,
      cheminFichier: json['cheminFichier'] as String,
      dureeSecondes: json['dureeSecondes'] as int?,
      dateEnregistrement: json['dateEnregistrement'] as String?,
      transcription: json['transcription'] as String?,
    );
  }
}
